src/HospitalLibrary/Allergens/AllergenRepository.cs

```diff
@@ -26,5 +26,11 @@ public Task<Allergen> GetOneByName(string name)
         {
             return _dataContext.Allergens.Where(a => a.Name.Equals(name)).FirstOrDefaultAsync();
         }
+
+        public List<int> GetAllergenIdsByPatient(int patientId)
```

Comment:
Instead of calling allergens from the allergens repository, you could have called include on patient and select only ids, but if this works its ok